### PR TITLE
Updated with publishing to SAR via git release

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

**Review by brinxmat — DISMISSED** [review]
I registered two bugs related to the travis config

File: template.yaml [hunk 1]
```diff
@@ -10,9 +10,9 @@ Metadata:
     Name: PublicationChannelRegister
     Description: Channel Registry for publications
     Author: Unit
-    SemanticVersion: 1.0.0
```
Comment (janbj) [inline template.yaml:13]: We don't use SemanticVersion anymore?

File: buildspec_publish_to_sar.yml [hunk 2]
```diff
@@ -6,17 +6,21 @@ phases:
...
   post_build:
     commands:
+      # Getting latest version tag from git
+      - APPLICATION_VERSION=$(curl -s  https://api.github.com/repos/BIBSYSDEV/$GIT_REPO/releases/latest | jq -r '.tag_name')
+      # Updating metadata.labels in template
+      - envsubst '${CODEBUILD_RESOLVED_SOURCE_VERSION},${GIT_REPO}' < sampackaged_raw.yaml > sampackaged.yaml
       # publishing to SAR
-      - sam publish --template sampackaged.yaml
-
+      - sam publish  --semantic-version $APPLICATION_VERSION  --template sampackaged.yaml
```
Comment (janbj) [inline buildspec_publish_to_sar.yml:23]: Maybe add a comment that we are setting SemanticVersion here and not in Metadata section.

File: buildspec_publish_to_sar.yml [hunk 3]
```diff
@@ -6,17 +6,21 @@ phases:
...
       # Use AWS SAM to build and package the application by using AWS CloudFormation
       - sam build
-      - sam package --template-file .aws-sam/build/template.yaml --s3-bucket $S3_BUCKET --output-template-file sampackaged.yaml --force-upload
+      - sam package --template-file .aws-sam/build/template.yaml --s3-bucket $S3_BUCKET --output-template-file sampackaged_raw.yaml
   post_build:
     commands:
+      # Getting latest version tag from git
+      - APPLICATION_VERSION=$(curl -s  https://api.github.com/repos/BIBSYSDEV/$GIT_REPO/releases/latest | jq -r '.tag_name')
+      # Updating metadata.labels in template
+      - envsubst '${CODEBUILD_RESOLVED_SOURCE_VERSION},${GIT_REPO}' < sampackaged_raw.yaml > sampackaged.yaml
```
Comment (janbj) [inline buildspec_publish_to_sar.yml:21]: These are both provided by the CodeBuild instance?

**Review by janbj — CHANGES_REQUESTED** [review]
I don't think the gradle wrapper has the correct permissions. Look at the failing jobs on Travis.

File: template.yaml (hunk 1 above)
Comment (svenngjeto) [reply template.yaml:13]: Moved to command line, but we could  use envsubst and keep it in template

File: buildspec_publish_to_sar.yml (hunk 2 above)
Comment (svenngjeto) [reply buildspec_publish_to_sar.yml:23]: Will do!

File: buildspec_publish_to_sar.yml (hunk 3 above)
Comment (svenngjeto) [reply buildspec_publish_to_sar.yml:21]: CODEBUILD_RESOLVED_SOURCE_VERSION is set by codebuild, GIT_REPO is manually set from the build codepipeline as environment variable.

**Review by janbj — APPROVED** [review]

**Pull request merged**